Makefile

```diff
@@ -269,6 +272,27 @@ ALL_PROG_OBJ	     := $(PROG_COMMON_OBJ) $(NONTEST_PROG_OBJ) \
 programs/config.h:scripts/detect.sh .build-config
 	$(QUIET_GEN) CC="$(CC)" CFLAGS="$(PROG_CFLAGS)" $< > $@
 
+# generate a pkg-config support file. requires GNU Make.
+define PKGCONFIG
```

Comment:
This is a weird approach.  Why not create a `libdeflate.pc.in` template file, and process it with `sed` to do the needed substitutions?  FWIW, that is the approach taken in another project I maintain: https://git.kernel.org/pub/scm/linux/kernel/git/ebiggers/fsverity-utils.git/tree/lib/libfsverity.pc.in

Reply:
sure, that's reasonable too (that's what i meant by "otherwise we need do some sed"). i don't feel that one's particularly better than this (the real problem is that flags passed on the Make command line don't force a rebuild, which would be true in either case), but your project, your rules.

Reply:
it does appear to work as expected btw

```
prefix=/usr/local
exec_prefix=${prefix}
includedir=${prefix}//usr/local/include
libdir=${prefix}/usr/local/lib

Name: libdeflate
Description: Fast implementation of DEFLATE, zlib, and gzip
Version: 1.8
Libs: -L${libdir} -ldeflate
Cflags: -I${includedir}
```

Reply:
> the real problem is that flags passed on the Make command line don't force a rebuild, which would be true in either case

Well that is what the `.build-config` file is for.  Just make the `libdeflate.pc` target depend on that, in addition to `libdeflate.pc.in`.  And add `PREFIX`, `LIBDIR`, and `INCDIR` to `.build-config`.

Reply:
Actually I forgot -- that won't work because those variables may not be defined until `make install` time.  So `libdeflate.pc` needs to be generated by `make install`.  See the other project I linked to that has a very similar Makefile -- that is what it does.